### ISSUE TITLE
IndexError in engine catches two cases

### ISSUE THREAD

**Issue by pathunstrom** [pathunstrom/pursuedpybear#5]
It's built to handle the event of an empty queue, but also catches an empty scene stack.

Break up the exception handling.

**Comment by pathunstrom** [comment]
```
try:
    cur_event = event_queue.pop()
    publisher.publish(cur_event)
    scenes[-1].publish(cur_event)
except IndexError:
    tick()
```

**Comment by pathunstrom** [comment]
Fixed with [commit 6b9ee9d746e13ca8eba22aca4aff3e29f318bf6e](https://github.com/pathunstrom/pursuedpybear/commit/6b9ee9d746e13ca8eba22aca4aff3e29f318bf6e)
